src/app/public/modules/action-buttons/action-buttons.component.html

```diff
@@ -8,20 +8,36 @@
   </div>
   <sky-row>
     <nav class="stache-action-buttons-nav-container">
-      <sky-column screenSmall="6" screenMedium="4" screenLarge="3" *ngFor="let route of filteredRoutes">
-        <a stacheRouterLink="{{route.path}}" fragment="{{route.fragment}}" class="stache-action-button-override sky-action-button sky-btn-default sky-rounded-corners">
-          <div>
-            <sky-action-button-icon *ngIf="route.icon" iconType="{{route.icon}}">
-            </sky-action-button-icon>
-            <sky-action-button-header>
-              {{route.name}}
-            </sky-action-button-header>
-          </div>
-          <sky-action-button-details>
-            {{route.summary}}
-          </sky-action-button-details>
-        </a>
-      </sky-column>
+      <ng-container *ngFor="let route of filteredRoutes">
+
+        <ng-container *ngIf="route.restricted else nonRestricted">
+          <ng-container *skyRestrictedView>
+            <ng-container *ngTemplateOutlet="actionButton"></ng-container>
+          </ng-container>
+        </ng-container>
```

Comment:
☝️ I'm not crazy about using these nested `ng-container`s, but we can't use more than one structural directive on an element, and I didn't want to mess with the HTML structure. If you have a better suggestion on how to handle this, I'm all ears.

Reply:
would some empty `div`s suffice? i think that beats the containers. and this dom structure already is suspect b/c it doesn't have the `sky-action-button-container` which has caused some recent headaches.

Reply:
Ah, good point about the `sky-action-button-container`. I didn't even think to check and see if this template was doing it right in the first place 😆 . I'm not a big fan of empty div's if they don't serve a purpose - that's kind of why I went with the `ng-container`s because they give a nice way to add logic to the template without running the risk of `div`-itis. However, I'll take a look at updating the structure of these action buttons and a better avenue might reveal itself.
More to come on this soon...

Reply:
@Blackbaud-ColbyWhite I spent some more time on this. Even after using the proper action button structure, I still need to wrap it with the restricted view directive, so we're kind of in the same boat.

I did have another thought - this much editing to the template made me wonder if this is even a good fit (architecturally). What if the `SkyRestrictedViewAuthService` exposed its service so we could use it in the typescript files (instead of the template)? That way we could filter out the routes that are restricted before they even make it to the template. What do you think?

Reply:
🤷 i'm struggling to visualize what that would look like. sounds reasonable.

Reply:
All right, I've pulled in the latest from `skyux-lib-restricted-view` and am now using the exposed `isAuthenticated` property. All the logic has been moved to the typescript file.